### ISSUE TITLE
protobuf-maven-plugin Config Causes Protoc Execution Error on Linux Systems

### ISSUE THREAD

**Issue by llj37825** [grpc/grpc-java#11236]
I am facing an issue with the protobuf-maven-plugin in a Spring Boot project that integrates gRPC. The plugin is configured as follows in the pom.xml:

<plugin>
    <groupId>org.xolstice.maven.plugins</groupId>
    <artifactId>protobuf-maven-plugin</artifactId>
    <version>0.6.1</version>
    <configuration>
        <protocArtifact>com.google.protobuf:protoc:3.7.1:exe:${os.detected.classifier}</protocArtifact>
        
    </configuration>
    
</plugin>
The problem arises when the CI/CD pipeline, which runs on a Linux environment, attempts to compile and package the application. The error message indicates that it's trying to execute a .exe file, which is not compatible with Linux:

An error occurred while invoking protoc: Error while executing process.: Cannot run program "protoc-3.7.1-linux-x86_64.exe": error=2, No such file or directory -> [Help 1]
Since Linux cannot execute .exe files, the build fails. How can this issue be resolved to ensure that the protobuf-maven-plugin correctly identifies and uses the appropriate protoc binary for Linux during the process?

**Comment by kannanjgithub** [comment]
The exe extension is a file format for plugins, as it is a plugin for the protoc compiler and it works on Linux also. 
Are you able to invoke the plugin from the command line directly using protoc and providing the downloaded plugin path, for ex., with 

`protoc --plugin=protoc-gen-grpc-java=/usr/local/homej/Downloads/protoc-gen-grpc-java-1.64.0-linux-x86_64.exe   --grpc-java_out=gen-files/ --proto_path=path/to/my/test.proto`

I found a similar reported error for Alpine Linux container [here](https://github.com/xolstice/protobuf-maven-plugin/issues/23). Can you confirm which flavor of Linux you are using?

**Comment by ejona86** [comment]
protoc-3.7.1 is really old, from 2019. But I don't think that is likely to be the cause. Can you try using our examples and seeing if they build? https://github.com/grpc/grpc-java/blob/master/examples/pom.xml

FWIW, nothing here is actually grpc-java, as we don't maintain the maven plugin nor protoc itself.

**Comment by llj37825** [comment]
> protoc-3.7.1 is really old, from 2019. But I don't think that is likely to be the cause. Can you try using our examples and seeing if they build? https://github.com/grpc/grpc-java/blob/master/examples/pom.xml
> 
> FWIW, nothing here is actually grpc-java, as we don't maintain the maven plugin nor protoc itself.

System Information:

Linux bd937df84756 5.4.262-1.el7.elrepo.x86_64 #1 SMP Wed Nov 29 00:56:30 EST 2023 x86_64 Linux
Issue:

To verify the existence of the protoc file, I entered the container and found the file protoc-3.17.2-linux-x86_64.exe (I switched the version). Even though the file exists, when I navigate to its directory and execute the command:

bash
protoc-3.17.2-linux-x86_64.exe --version
I still get an error saying the file does not exist. Here are the steps I followed:

Download and unzip the file:

bash
curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v3.17.2/protoc-3.17.2-linux-x86_64.zip
unzip protoc-3.17.2-linux-x86_64.zip -d protoc3

bash
mv protoc3/bin/* /usr/local/bin/
mv protoc3/include/* /usr/local/include/
chmod +x /usr/local/bin/protoc

After performing the above steps, when I execute the command:

bash
protoc --version
I receive the error that the protoc file does not exist.
